---
title: 'Microbiota profiling in consideration of varying P-levels: Import & Normalization'
author: "Claudia Probst based on Jan Waelchli and Charlotte Joller"
date: "`r Sys.Date()`"
geometry: margin=2cm
output:
  pdf_document:
    toc: yes
    toc_depth: 3
editor_options: 
  chunk_output_type: console
---


\pagebreak

# Experimental Setup

We’re investigating the responses of *Arabidopsis thaliana* to varying levels of P-fertilizer supply in presence or absence of a synthetic bacterial commmunity composed of 8 taxonomically diverse strains (SynCom). The SynCom is coposed of: 

* F112 Pseudomonas
* F247 Pseudomonas
* F242 Labrys
* F233 Nocardioides
* F201 Ensifer
* F004 Domibacillus
* F042 Variovorax
* F125 Rhodanobacter
* (F041 Cupriavidus)


*Arabidopsis thaliana* was watered with nutrient solution differeing in its P-content: 0.03mM, 1mM, 5mM K2HPO4. In a P-limited envionment, we suppose that more beneficial microbes are recruited and help the plant to take up nutrients efficiently. 

Treatments were as followed:

* low P control (0.03 mM K2HPO4) --> 5 replicates
* low P + SynCom9 (0.03 mM K2HPO4 + SynCom9) --> 10 replicates
* medium P control (1mM K2HPO4) --> 5 replicates
* medium P + SynCom9 (1mM K2HPO4) --> 10 replicates
* high P cntrol (5mM K2HPO4) --> 5 replicates
* high P + SynCom9 (5mM K2HPO4) --> 10 replicates


For analysis of the microbiome profile we are interested to compare:

* substrate vs root -> type
* different treatments --> high vs medium vs low
* SynCom9 vs control -> SynCom9 yes/no


```{r setup, include=FALSE, echo=F, message=F, warning=F}

#clear the objects from memory
rm(list=ls())

#knitr settings
knitr::opts_chunk$set(echo=TRUE, fig.align="center")
options(tinytex.verbose = TRUE)

#set seed<
set.seed(100)

#set paths
paths <- list(wd = c("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results"),
              DESIGN = c("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/BS03_design/design.xlsx"),
              bASV_table = c("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/BS03_output/bacteria/100/bacteria_COUNT100.tab"),
              btaxa = c("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/BS03_output/bacteria/100/bacteria_TAXA100.tab"),
              btrack = c("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/BS03_output/sequence_tracking/bacteria_sample_track.xlsx"),
              functions="~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/functions/"
              )


## set source to file location
if (!require(rstudioapi)){install.packages("rstudioapi")}; library(rstudioapi)
setwd(dirname(getActiveDocumentContext()$path))

## load functions

#source("4_R_functions/libraries.R")
#libraries()

source(paste0(paths$functions,"functions.R"))
functions(path=paths$functions)
#libraries()
## installs (if necessary) and loads libraries
library(conflicted)
conflict_prefer("filter", "dplyr")
conflict_prefer("rename", "dplyr")
conflict_prefer("select", "dplyr")
conflict_prefer("mutate", "dplyr")
conflict_prefer("arrange", "dplyr")

library(phyloseq)
library(vegan)
library(ALDEx2)
#library(ANCOMBC)
#library(Maaslin2)
#library(metagenomeSeq)
library(igraph)
library(sciplot)
library(ggplot2)
library(plyr)
library(pander)
library(reshape)
library(knitr)
library(gplots)
library(cowplot)
library(Hmisc)
library(stringr)
library(dplyr)
library(emmeans)
library(multcomp)
library(multcompView)
library(forcats)
library(tibble)
library(tidyr)
library(tidyverse)
library(sylly)
library(readxl)
library(phyloseq)
library(MetBrewer)
```

```{r import design, echo=F, message=F, warning=F}


#import design file and tidy it
DESIGN <- read_excel(paths$DESIGN) %>%
  select(!c(taxa, run_nbr, f_microbe_primer_ID, f_microbe_primer_seq, r_microbe_primer_ID, r_microbe_primer_seq, f_plant_primer_ID, f_plant_primer_seq, r_plant_primer_ID, r_plant_primer_seq, bc_seq)) #remove variables that are not helpful for analysis


#factors
DESIGN$type <- factor(DESIGN$type, levels=c("roots", "sorbix"))
DESIGN$SynCom9 <- factor(DESIGN$SynCom9, levels = c("yes", "no"))
DESIGN$treatment <- factor(DESIGN$treatment, levels = c("high P", "medium P", "low P"))

DESIGN <- DESIGN %>% 
  mutate(cols_type=if_else(type=="sorbix", "#88a0dc", "#f9d14a")) %>% 
  mutate(cols_SynCom9=if_else(SynCom9=="yes", "#f94aab", "#4ab7f9")) %>% 
  mutate(cols_treatment=if_else(treatment=="high P","aquamarine3", "darkgoldenrod1"))%>%
  mutate(cols_treatment=if_else(treatment=="medium P", "cornflowerblue", cols_treatment)) 
  
## collapsed color vectors
temp <- data.frame(DESIGN$type, DESIGN$cols_type)
temp <- plyr::ddply(temp, .variables="DESIGN.cols_type", .fun=unique)
level_cols_type <- as.character(temp[,2])
names(level_cols_type) <- temp[,1]

temp <- data.frame(DESIGN$SynCom9, DESIGN$cols_SynCom9)
temp <- plyr::ddply(temp, .variables="DESIGN.cols_SynCom9", .fun=unique)
level_cols_SynCom9 <- as.character(temp[,2])
names(level_cols_SynCom9) <- temp[,1]

temp <- data.frame(DESIGN$treatment, DESIGN$cols_treatment)
temp <- plyr::ddply(temp, .variables="DESIGN.cols_treatment", .fun=unique)
level_cols_treatment <- as.character(temp[,2])
names(level_cols_treatment) <- temp[,1]
rm(temp)
```

```{r DAT import, echo=F, message=F, warning=F, include=F}

#import ASV table
DAT <- read.delim(paths$bASV_table, header=T, row.names=1, sep="\t")
DAT <- t(DAT) # transpose

# simplify names
colnames(DAT) <- gsub("BS03_", "", colnames(DAT))
rownames(DAT) <- gsub("OTU","ASV", rownames(DAT)) #the rows were named OTUs even though they are ASVs really (100% sequence identity).

#sort and remove input and cneg samples
order <- match(DESIGN$AccessArray, colnames(DAT))
bDAT <- as.data.frame(DAT[, order])

#split make input and cneg DAT
order <- match(DESIGN$AccessArray, colnames(DAT))
inputDAT <- as.data.frame(DAT[, order])

#Check for samples that had 0s for all ASVs (nothing sequenced)
bDAT0 <- bDAT %>% select_if(colSums(.) == 0) # No sample had zero reads for all ASVs
#remove ASVs with 0 seqs
#bDAT <- DAT[rowSums(DAT) >0,]

#Turn bDAT into a dataframe
bDAT <- as.data.frame(bDAT)

#remove 

rm(DAT)
rm(bDAT0)
rm(order)


```

```{r bTAX import, echo=F, message=F, warning=F, include=F}

### BACTERIA ###
#import taxonomy table and SynCom similarities file
TAX <- read.table(paths$btaxa, row.names=1, sep="\t", blank.lines.skip = FALSE)
SynSim <- read_excel("similarities.xlsx")

#rename
colnames(TAX) <- c("kingdom", "phylum", "class", "order", "family", "genus")
rownames(TAX) <- gsub("OTU","ASV", rownames(TAX)) #the rows were named OTUs even though they are ASVs really (100% sequence identity).
SynSim$OTU <- gsub("OTU","ASV", SynSim$OTU)
TAX[is.na(TAX)] <- "unassigned"
SynSim[is.na(SynSim)] <- "unassigned"

#add ASV_ID to TAXonomy file
TAX$ASV_ID <- rownames(TAX)

#merge SynComStrain info to TAX file
TAX <- TAX %>% 
  left_join(SynSim, by = join_by(ASV_ID == OTU)) %>% 
  column_to_rownames(var = "ASV_ID") %>% 
  mutate(ASV_ID=rownames(TAX)) %>% 
  mutate(SynComStrain=if_else(kingdom=="Eukaryota", "Host", SynComStrain))

#remove OTU with 0 seqs
TAX <- TAX[rownames(TAX) %in% rownames(bDAT),]

# define ASVs for removal
#r1 <- -which(TAX$kingdom=="Eukaryota")
r2 <- -which(TAX$phylum=="Cyanobacteria")
r3 <- -which(TAX$family=="Mitochondria")
r4 <- -which(is.na(TAX)) #rows all na
#ASVs_to_remove <- c(r1,r2,r3,r4)
ASVs_to_remove <- c(r2,r3,r4)
if(length(ASVs_to_remove)>0){
  TAX <- TAX[ASVs_to_remove ,]
  bDAT <- bDAT[rownames(TAX),]
}

#create a bTAXonomy label made up of family and genus
TAX <- TAX %>% 
  unite("label", 5:6, sep = "_", remove = FALSE)
#add Eukaryota to labels
TAX$label[TAX$kingdom == "Eukaryota"] <- "Host"


# vector of colors for abundant family and genuses
# will be used for graphs later
TAX$cols_famgen <- as.character(TAX$label)

try(TAX[ TAX$label=="Burkholderiaceae_Cupriavidus" , ]$label <- "#Pseudomonadaceae_Pseudomonas")

TAX$cols_famgen <- "gray87"
try(TAX[ TAX$label=="Pseudomonadaceae_Pseudomonas" , ]$cols_famgen <- "#4F447E")
try(TAX[ TAX$label=="Pseudomonadaceae_unassigned" , ]$cols_famgen <- "#baacd1")
try(TAX[ TAX$label=="Rhodanobacteraceae_Luteibacter" , ]$cols_famgen <- "#82438a")
try(TAX[ TAX$label=="Rhodanobacteraceae_unassigned" , ]$cols_famgen <- "#cfaec9")
try(TAX[ TAX$label=="Burkholderiaceae_Variovorax" , ]$cols_famgen <- "#237E91")
#try(TAX[ TAX$label=="Burkholderiaceae_Cupriavidus" , ]$cols_famgen <- "#2A6494")
try(TAX[ TAX$label=="Rhizobiaceae_Ensifer" , ]$cols_famgen <- "#6fb04e")
try(TAX[ TAX$label=="Rhizobiaceae_unassigned" , ]$cols_famgen <- "#9ac883")
try(TAX[ TAX$label=="Labraceae_Labrys" , ]$cols_famgen <- "#205131")
try(TAX[ TAX$label=="Nocardioidaceae_Nocardioides" , ]$cols_famgen <- "#D55474")
try(TAX[ TAX$label=="Planococcaceae_Domibacillus", ]$cols_famgen <- "#FFD162")
try(TAX[ TAX$label=="Planococcaceae_unassigned", ]$cols_famgen <- "#F78C09")
try(TAX[ TAX$label=="Host", ]$cols_famgen <- "#005E10")

try(TAX[ TAX$label=="Acetobacteraceae_Roseomonas", ]$label <- "unassigned")
try(TAX[ TAX$label=="Pseudonocardiaceae_Saccharothrix", ]$label <- "unassigned")
try(TAX[ TAX$label=="Caulobacteraceae_Brevundimonas", ]$label <- "unassigned")
try(TAX[ TAX$label=="Devosiaceae_Devosia", ]$label <- "unassigned")
try(TAX[ TAX$label=="Dietziaceae_Dietzia", ]$label <- "unassigned")
try(TAX[ TAX$label=="Enterobacteriaceae_Chania", ]$label <- "unassigned")
try(TAX[ TAX$label=="Enterobacteriaceae_Citrobacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Enterobacteriaceae_Enterobacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Nocardiaceae_Rhodococcus", ]$label <- "unassigned")
try(TAX[ TAX$label=="Nocardiaceae_Williamsia", ]$label <- "unassigned")
try(TAX[ TAX$label=="Nocardioidaceae_Aeromicrobium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Microbacteriaceae_Microbacterium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Micrococcaceae_Pseudarthrobacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Microbacteriaceae_Yonghaparkia", ]$label <- "unassigned")
try(TAX[ TAX$label=="Mycobacteriaceae_Mycobacterium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Nakamurellaceae_Nakamurella", ]$label <- "unassigned")
try(TAX[ TAX$label=="Paenibacillaceae_Paenibacillus", ]$label <- "unassigned")
try(TAX[ TAX$label=="Beijerinckiaceae_Methylobacterium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Beijerinckiaceae_Bosea", ]$label <- "unassigned")
try(TAX[ TAX$label=="Beijerinckiaceae_Methylorosula", ]$label <- "unassigned")
try(TAX[ TAX$label=="Burkholderiaceae_Massilia", ]$label <- "unassigned")
try(TAX[ TAX$label=="Burkholderiaceae_Ramlibacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Burkholderiaceae_Caenimonas", ]$label <- "unassigned")
try(TAX[ TAX$label=="Burkholderiaceae_Noviherbaspirillum", ]$label <- "unassigned")
try(TAX[ TAX$label=="Hyphomicrobiaceae_Hyphomicrobium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Streptococcaceae_Streptococcus", ]$label <- "unassigned")
try(TAX[ TAX$label=="Streptomycetaceae_Streptomyces", ]$label <- "unassigned")
try(TAX[ TAX$label=="Streptomycetaceae_unassigned", ]$label <- "unassigned")
try(TAX[ TAX$label=="Intrasporangiaceae_Knoellia", ]$label <- "unassigned")
try(TAX[ TAX$label=="Intrasporangiaceae_Janibacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Methylophilaceae_Methylotenera", ]$label <- "unassigned")
try(TAX[ TAX$label=="Xanthobacteraceae_Rhodopseudomonas", ]$label <- "unassigned")
try(TAX[ TAX$label=="Xanthobacteraceae_Bradyrhizobium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Xanthobacteraceae_unassigned", ]$label <- "unassigned")
try(TAX[ TAX$label=="Xanthomonadaceae_Lysobacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Xanthomonadaceae_unassigned", ]$label <- "unassigned")
try(TAX[ TAX$label=="Solirubrobacteraceae_Patulibacter", ]$label <- "unassigned")
try(TAX[ TAX$label=="Rhizobiaceae_Aureimonas", ]$label <- "unassigned")
try(TAX[ TAX$label=="Rhizobiaceae_Neorhizobium", ]$label <- "unassigned")
try(TAX[ TAX$label=="Rhodanobacteraceae_Chiayiivirga", ]$label <- "unassigned")
try(TAX[ TAX$label=="unassigned_unassigned", ]$label <- "unassigned")

# vector of colors for phyla present (and classes for Proteobacteria)
TAX$label_phyla <- as.character(TAX$phylum)
# create separate bTAXonomy label specifying classes of Proteobacteria
TAX$class <- as.factor(TAX$class)
try(TAX[ TAX$class=="Alphaproteobacteria", ]$label_phyla <- "Alphaproteobacteria")
try(TAX[ TAX$class=="Betaproteobacteria", ]$label_phyla <- "Betaproteobacteria")
try(TAX[ TAX$class=="Gammaproteobacteria", ]$label_phyla <- "Gammaproteobacteria")
try(TAX[ TAX$class=="Deltaproteobacteria", ]$label_phyla <- "Deltaproteobacteria")
try(TAX[ TAX$kingdom=="Eukaryota", ]$label_phyla <- "Host")
try(TAX[ TAX$kingdom=="unassigned", ]$label_phyla <- "unassigned")

TAX$label_phyla <- as.factor(TAX$label_phyla)
# vector of colors for abundant phyla (and classes for Proteobacteria)
# will be used for graphs later
TAX$cols_phyla <- as.character(TAX$label_phyla)
TAX$cols_phyla <- "gray87"
try(TAX[ TAX$label_phyla=="Alphaproteobacteria" , ]$cols_phyla <- "#6fb04e")
try(TAX[ TAX$label_phyla=="Gammaproteobacteria" , ]$cols_phyla <- "#4F447E")
try(TAX[ TAX$label_phyla=="Actinobacteria" , ]$cols_phyla <- "#D55474")
try(TAX[ TAX$label_phyla=="Firmicutes" , ]$cols_phyla <- "#F4C00D")
try(TAX[ TAX$label_phyla=="Host" , ]$cols_phyla <- "#005E10")

# vector of colors for the SynCom
TAX$cols_SynCom <- as.character(TAX$SynComStrain)
TAX$cols_SynCom <- "gray87"
#Cupri, blau
try(TAX[ TAX$SynComStrain=="F041" , ]$SynComStrain <- "F247")


#Pseudo
try(TAX[ TAX$SynComStrain=="F247" , ]$cols_SynCom <- "#454277")
try(TAX[ TAX$SynComStrain=="F112" , ]$cols_SynCom <- "#6e69ac")
#Rhodano
try(TAX[ TAX$SynComStrain=="F125" , ]$cols_SynCom <- "#82438a")
#Vario, türkis
try(TAX[ TAX$SynComStrain=="F042" , ]$cols_SynCom <- "#237E91")
#try(TAX[TAX$SynComStrain=="F041", ]$cols_SynCom <- "#454277") #Cupriavidus was contaminated by F247 by mistake
#Labrys
try(TAX[ TAX$SynComStrain=="F242" , ]$cols_SynCom <- "#205131")
#Rhi_Ensifer
try(TAX[ TAX$SynComStrain=="F201" , ]$cols_SynCom <- "#6fb04e")
#Domi
try(TAX[ TAX$SynComStrain=="F004" , ]$cols_SynCom <- "#FFD162")
#Actino
try(TAX[ TAX$SynComStrain=="F233" , ]$cols_SynCom <- "#D55474")
#Host
try(TAX[ TAX$SynComStrain=="Host" , ]$cols_SynCom <- "#005E10")


## collapsed color vectors for each level
temp <- data.frame(TAX$label_phyla, TAX$label, TAX$cols_famgen)
temp <- temp %>% 
  arrange(TAX.label_phyla)
temp <- plyr::ddply(temp, .variables="TAX.label", .fun=unique)
TAX_level_cols_famgen <- as.character(temp[,3])
names(TAX_level_cols_famgen) <- temp[,2]

temp <- data.frame(TAX$label_phyla, TAX$label, TAX$cols_phyla)
temp <- temp %>% 
  arrange(TAX.label_phyla) %>% 
  distinct(TAX.label_phyla, TAX.cols_phyla)
TAX_level_cols_phyla <- as.character(temp[,2])
names(TAX_level_cols_phyla) <- temp[,1]

temp <- data.frame(TAX$label_phyla, TAX$SynComStrain, TAX$cols_SynCom)
temp <- temp %>% 
  arrange(TAX.label_phyla) %>% 
  distinct(TAX.SynComStrain, TAX.cols_SynCom)
TAX_level_cols_SynCom <- as.character(temp[,2])
names(TAX_level_cols_SynCom) <- temp[,1]

# remove no longer used files
rm(temp)

```

### Figure 1.1 \| Rarefaction

In a first step we conduct a rarefaction plot to make sure that we have sequenced deep enough.

\vspace{5mm}

```{r rarefaction, eval=T, echo=F, message=F, warning=F, fig.height=4, fig.width=15}

#BACTERIA

#rarefy
pdf("/dev/null") #prevents drawing of standard rarefaction plot (we will create our own with colors)
brarefaction <- rarecurve(t(bDAT), step=200, 7000, label = FALSE) 
names(brarefaction) <- DESIGN$AccessArray
invisible(dev.off())

# long-transformation.
protox <- mapply(FUN = function(x, y) {
  mydf <- as.data.frame(x)
  colnames(mydf) <- "value"
  mydf$species <- y
  mydf$subsample <- attr(x, "Subsample")
  mydf
}, x = brarefaction, y = as.list(names(brarefaction)), SIMPLIFY = FALSE)

brarefaction_long <- do.call(rbind, protox)
rownames(brarefaction_long) <- NULL  # pretty

#add type roots vs sorbix
brarefaction_long$type <- NA
for (i in 1:nrow(brarefaction_long)) {
  brarefaction_long$type[i] <- as.character(DESIGN$type[DESIGN$AccessArray == brarefaction_long$species[i]])
}

#add SynCom 9 yes/no
brarefaction_long$SynCom9 <- NA
for (i in 1:nrow(brarefaction_long)) {
  brarefaction_long$SynCom9[i] <- as.character(DESIGN$SynCom9[DESIGN$AccessArray == brarefaction_long$species[i]])
}

#add treatments
brarefaction_long$treatment <- NA
for (i in 1:nrow(brarefaction_long)) {
  brarefaction_long$treatment[i] <- as.character(DESIGN$treatment[DESIGN$AccessArray == brarefaction_long$species[i]])
}

#plot
bac_rareplot1 <- ggplot(brarefaction_long, aes(x = subsample, y = value, group = species)) +
                  theme_bw() +
                  geom_line(aes(color=type))+
                  scale_color_manual(values = level_cols_type)+
                  xlab("Sequencing Depth")+
                  scale_x_continuous(limits=c(0,25000))+
                  ylab("Number of ASVs")+
                  ggtitle("Rarefaction Roots vs Sorbix") +
                  theme(legend.position = "bottom")
bac_rareplot2 <- ggplot(brarefaction_long, aes(x = subsample, y = value, group = species)) +
                  theme_bw() +
                  geom_line(aes(color=SynCom9))+
                  scale_color_manual(values = level_cols_SynCom9)+
                  xlab("Sequencing Depth")+
                  scale_x_continuous(limits=c(0,25000))+
                  ylab("Number of ASVs")+
                  ggtitle("Rarefaction SynCom9 yes/no")+
                  theme(legend.position = "bottom")
bac_rareplot3 <- ggplot(brarefaction_long, aes(x = subsample, y = value, group = species)) +
                  theme_bw() +
                  geom_line(aes(color=treatment))+
                  scale_color_manual(values = level_cols_treatment)+
                  scale_x_continuous(limits=c(0,25000))+
                  xlab("Sequencing Depth")+
                  ylab("Number of ASVs")+
                  ggtitle("Treatments (high P/medium P/low P)")+
                  theme(legend.position = "bottom")


#combine the two plots (library cowplot)
fig1.1 <- plot_grid(bac_rareplot1,
               bac_rareplot2 + labs(colour=""),
               bac_rareplot3 + labs(clour=""),
               labels=c("A","B", "C"),
               align = "hv",
               nrow=1)

fig1.1
#ggsave("fig1.1.png", height=4, width=15) 
# dev.off()


```
\vspace{5mm}

**Conclusion:** The samples were sequenced deep enough. It looks like the diversity is smaller in the Sorbix samples compared to the root samples. 

\vspace{5mm}

### Figure 1.2 \| Number of reads

As an additional control, we plot the amount of reads during each pipeline step. This allows us to see where we loose reads and if the samples from the different treatment groups behave similarly.

\vspace{5mm}

```{r track sequences, eval=T, echo=F, message=F, warning=F, , fig.height=10, fig.width=10}
#BACTERIA

#import
track <- read_xlsx(paths$btrack)
colnames(track)[1] <- "FLD"
track$FLD <- gsub("BS03_","",track$FLD)

#sort in order of Design
order <- match(DESIGN$AccessArray, track$FLD)
track <- track[order,]

#add seqs after removing eukaryota seqs
#track$noneukar <- colSums(bDAT)

#add genotype, arsen
track$type <- as.factor(DESIGN$type)
track$SynCom9 <- as.factor(DESIGN$SynCom9)
track$treatment <- as.factor(DESIGN$treatment)
track_long <- pivot_longer(track, cols= !(c("FLD", "type", "SynCom9", "treatment")), names_to = "steps", values_to = "reads")

track_long$type <- as.factor(track_long$type)
track_long$SynCom9 <- as.factor(track_long$SynCom9)
track_long$treatment <- as.factor(track_long$treatment)
track_long$steps <- factor(track_long$steps, levels=c("input", "filter_length", "filter_maxN","filter_maxEE", "denoisedF","denoisedR", "merged", "nonchim","noneukar"))
levels(track_long$steps)[1] <- "demultiplexed"

track_bac1 <- ggplot(track_long, aes(steps, reads, col=type, group=FLD))+
              #geom_jitter(alpha=0.1, width = 0.2)+
              geom_smooth(aes(), se=F, method="loess", formula="y~x")+
              theme_bw()+
              scale_color_manual(values = level_cols_type)+
              #facet_grid(~"Number of reads")+
              ggtitle("Reads tracking Roots vs Sorbix")


track_bac2 <- ggplot(track_long, aes(steps, reads, col=SynCom9, group=FLD))+
              #geom_jitter(alpha=0.1, width = 0.2)+
              geom_smooth(aes(), se=F, method="loess", formula="y~x")+
              theme_bw()+
              scale_color_manual(values = level_cols_SynCom9)+
              #facet_grid(~"Number of reads")+
              ggtitle("Reads tracking SynCom yes/no")


track_bac3 <- ggplot(track_long, aes(steps, reads, col=treatment, group=FLD))+
              #geom_jitter(alpha=0.1, width = 0.2)+
              geom_smooth(aes(), se=F, method="loess", formula="y~x")+
              theme_bw()+
              scale_color_manual(values = level_cols_treatment)+
              #facet_grid(~"Number of reads")+
              ggtitle("Reads tracking Treatments (high P/medium P/low P)")


#combine the two plots (library cowplot)
fig1.2 <- plot_grid(track_bac1,
               track_bac2 + labs(colour=""),
               track_bac3 + labs(colour=""),
               labels=c("A","B", "C"),
               align = "hv",
               ncol=1)

fig1.2
#ggsave("fig1.2.png", height=10, width=10) 
# dev.off()
```

\vspace{5mm}

**Conclusion:** We have less sorbix reads compared to roots. We loose consistently ± the same amount of reads when running the data in the pipeline. 
\pagebreak

# Description all data

## Sequencing depth

We show the sum, range and median over all samples.

```{r seq numbers bacteria, echo=F, message=F, warning=F}

df <- data.frame(removed_samples=NA, 
                 sum=NA, min=NA, max=NA, median=NA, ASVs=NA)

#show sorted seq numbers
ReadNbr <- data.frame(sort(colSums(bDAT),decreasing = T))
ReadNbr <- ReadNbr %>% 
  rename("read_number"=1)
#pander(ReadNbr, caption="Total Number of Reads per Sample")

#remove samples with very low seq numbers
threshold <-  1600
samples_to_remove <- names(which(colSums(bDAT) < threshold))
print(paste(length(samples_to_remove),"samples removed"))

pander((DESIGN %>% filter(AccessArray %in% samples_to_remove)), caption = "removed samples")

#print(paste(length(samples_to_remove_sand),"samples removed"))
df$removed_samples <- length(samples_to_remove)

#Remove low sequence samples from DESIGN and bDAT file
DESIGN <- DESIGN[!(DESIGN$AccessArray %in% samples_to_remove),]
bDAT_keep <- bDAT %>% 
  select(!(samples_to_remove))

#seq numbers
df$sum <- sum(colSums(bDAT_keep))
df$min <- min(colSums(bDAT_keep))
df$max <- max(colSums(bDAT_keep))
df$median <- median(colSums(bDAT))
df$ASVs <- nrow(bDAT)


#out
pander(df, use.hyphening = TRUE, split.cells = 5)


## define rarefication threshold 
#(number of reads from the sample with the lowest seq-depth)
rare_bac <- min(colSums(bDAT_keep))
rare_bac <- floor(rare_bac / 100) *100 #round to the next lower 100


```
\vspace{5mm}

**Conclusion:** The number of reads per sample ranges from max = 37930 to min ± 1603 reads. The samples with less reads than 1500 is one root sample without SynCom9 and 3 Sorbix samples with medium P. I would intuitively set the threshold of minimum reads/sample to 1600 and trash the samples with less reads. With this threshold, 4 samples are "trashed". The rarefaction threshold was defined by the lowest number of reads of non-trashed samples = 1603



We compare the eukaryotic ASVs to the At.GI sequence and remove ASVs which are less similar than 99%

```{r At.GI amplicon, echo=F, warning=F, message=F}

library("Biostrings")
library("DECIPHER")

At.GI <- Biostrings::readDNAStringSet("At.GI_amplicon.fasta")
ISO <- Biostrings::readDNAStringSet("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/BS03_output/bacteria/100/bacteria_SEQ100.fasta")

### MAPPING ###

#separate Eukaryotes and Bacteria
ASV_euk <- rownames(TAX)[TAX$kingdom=="Eukaryota"]
DAT_euk <- bDAT_keep[rownames(bDAT_keep) %in% ASV_euk,]
ASV_bac <- rownames(TAX)[TAX$kingdom=="Bacteria"]
DAT_bac <- bDAT_keep[rownames(bDAT_keep) %in% ASV_bac,]

#convert DNAStringSet as data frame
At.GI_df <- as.data.frame(At.GI)
colnames(At.GI_df) <- "seqs"

ISO_df <- as.data.frame(ISO)
colnames(ISO_df) <- "seqs"
rownames(ISO_df) <- gsub("OTU","ASV", rownames(ISO_df))

#subset to Eukaryota ASVs
ISO_df <- ISO_df[rownames(ISO_df) %in% ASV_euk,, drop=F]

#create new data.frame to save distances
distances <- data.frame(matrix(NA, nrow = nrow(At.GI_df), ncol = nrow(ISO_df)), row.names = rownames(At.GI_df))
colnames(distances) <- rownames(ISO_df)

#mapping function
map <- function(At.GI_seq) {
  seqs_to_map <- data.frame(seqs=c(as.character(At.GI_seq),as.character(ISO_df$seqs[i]))) #create data.frame with At.GI and ISO to map
  seqs_to_map <- Biostrings::DNAStringSet(seqs_to_map$seqs) #convert to DNA string set
  seqs_aln <- DECIPHER::AlignSeqs(seqs_to_map, processors = 1, verbose = F) #align, processors = NULL uses the max available number of processors
  dist_all <- DECIPHER::DistanceMatrix(seqs_aln, processors = 1, verbose = F) #similarity table
  return(dist_all[1,2])
}

#calculate distances
#this loop may take a while
for(i in 1:nrow(ISO_df)) {
  #loop over each At.GI
  distances[,i] <- apply(At.GI_df, 1, map) #mapping and save values in dist data.frame
  #print(paste0("seq ", i, "/",  nrow(ISO_df), " mapped"))
}

#create similarity- and top-hits table
similarities <- 1 - distances #similarity

#add OTU abundance and plot
similarities <- rbind(similarities, n_seqs=rowSums(DAT_euk))
pander(t(similarities))

#remove OTUs
ASVs_to_remove <- colnames(similarities)[similarities[1,]<0.99]
bDAT <- bDAT_keep[!(rownames(bDAT_keep) %in% ASVs_to_remove),]
ASVs_to_remove <- names(which(colSums(bDAT_keep) == 0))
bDAT <- bDAT_keep[,!(colnames(bDAT_keep) %in% ASVs_to_remove)]
TAX <- TAX[!(rownames(TAX) %in% ASVs_to_remove),]

```
**Conclusion** ASV1 and ASV 143 match perfectly to the gene sequence of *A. thaliana* *Gigantea*. Both show a similarity of 1. 
\vspace{5mm}

\pagebreak

### Figure 2 \| Sequencing depth

\vspace{5mm}

```{r Figure 2, echo=F, warning=F, message=F, fig.height=10, fig.width=15}

df <- data.frame(col_sum=colSums(bDAT_keep),colnames=colnames(bDAT_keep), type=DESIGN$type, SynCom9=DESIGN$SynCom9, treatment=DESIGN$treatment)

ylim2=boxplot.stats(df$col_sum)$stats[c(1, 5)]

type.labs <- c("roots", "sorbix")
names(type.labs) <- c("roots", "sorbix")
SynCom9.labs <- c("+ SynCom9", "- SynCom9")
names(SynCom9.labs) <- c("yes", "no")


## boxplot
seq_nr1 <- df%>%
              ggplot(aes(x=type, y=col_sum, fill=treatment)) + 
              geom_boxplot(position=position_dodge2(width=0.75, preserve="single"),outlier.colour = NA) + 
              geom_jitter(size=1, position=position_jitterdodge(jitter.width=0.2,dodge.width = 0.75))+
              theme_bw() +
              theme(legend.position="none", 
                    axis.text.x=element_blank(), 
                    axis.title.x = element_text(size = 10, hjust = 0.5),
                    axis.title.y = element_text(size = 10, hjust = 0.5),
                    axis.text.y = element_text(size = 8, color = "black")) +
              ylab("seq") +
              scale_fill_manual(values=level_cols_treatment) +
              coord_cartesian(ylim=c(ylim2[1],ylim2[2]*1.4)) +
              facet_wrap(type~treatment, scales='free_x', labeller = labeller(type = type.labs, SynCom9 = SynCom9.labs, size=14, .multi_line = FALSE))+

              ggtitle("Sequencing depth")

seq_nr2 <- df%>%
              ggplot(aes(x=SynCom9, y=col_sum, fill=treatment)) + 
              geom_boxplot(position=position_dodge2(width=0.75, preserve="single"),outlier.colour = NA) + 
              geom_jitter(size=1, position=position_jitterdodge(jitter.width=0.2,dodge.width = 0.75))+
              theme_bw() +
              theme(legend.position="none", 
                    axis.text.x=element_blank(), 
                    axis.title.x = element_text(size = 10, hjust = 0.5),
                    axis.title.y = element_text(size = 10, hjust = 0.5),
                    axis.text.y = element_text(size = 8, color = "black")) +
              ylab("seq") +
              scale_fill_manual(values=level_cols_treatment) +
              coord_cartesian(ylim=c(ylim2[1],ylim2[2]*1.4)) +
              facet_wrap(SynCom9~treatment, scales='free_x', labeller = labeller(type = type.labs, SynCom9 = SynCom9.labs, size=14, .multi_line = FALSE))+

              ggtitle("Sequencing depth")

## barplot
df <- df[order(df$col_sum, decreasing = T),]

df$x <- c(1:nrow(df))

samples <-  ggplot(df, aes(x=x, y=col_sum, fill=treatment))+
                geom_bar(aes(width=1), size=0.1, stat = "identity", col="white", lty=1)+
                scale_fill_manual(values=level_cols_treatment) +
                xlab("samples")+
                ylab("seq / sample")+
                #facet_grid(~ scale, scales='free_x', space="free")+
                #geom_text(aes(label=colnames), angle=90, hjust=3, size=2)+
                theme_bw() +
                theme(axis.text.x=element_blank(), axis.ticks.x=element_blank(),
                      strip.background = element_blank(), strip.text.x = element_blank(), legend.position = "none")

#combine the two plots
fig2 <-cowplot::plot_grid(seq_nr1,
               seq_nr2 + labs(colour=""),
               align = "hv",
               rel_heights = c(10,10),
               nrow=1)

fig2 <- cowplot::plot_grid(fig2,
               samples + labs(colour=""),
               align = "hv",
               rel_heights = c(15,10),
               rel_widths = c(10,10),
               nrow=2)


#add legend
legend <- get_legend(seq_nr1 + theme(legend.position="bottom") + labs(fill="treatment"))
fig2<- plot_grid(fig2, legend, nrow=2, rel_heights=c(1, .1))
fig2

#ggsave("fig2.png", height=10, width=15) 

rm(df)
```
\vspace{5mm}

**Conclusion:** The number of reads differ between the different treatments in roots sample and in the + SynCom9 samples. We have more variation in the reads with low P. In Sorbix and the - SynCom9 samples we have a similar pattern when comparing the different treatments. 

\pagebreak

# Data normalization

## Asymptotic Kruskal-Wallis Test

To decide on how to normalize the data we follow the recommendation of Weiss et al. (2017, Microbiome Journal) and inspect whether there are differences in sequencing depths between the different sample types using the non-parametric Kruskal-Wallis Test.

```{r seq numbers per group bacteria, echo=F, message=F, warning=F}

#BACTERIA
group <- as.factor(paste(DESIGN$type, DESIGN$treatment,sep="_"))
sample_depth <- colSums(bDAT_keep)
DAT_krusk <- kruskal.test(sample_depth ~ group)  # library(coin)
print("Kruskal-Wallis Test")
DAT_krusk
```

\vspace{5mm}

**Conclusion:** The number of reads per sample are not significantly different between sample groups. We rarefy the data to normalize for differences. The rarefaction threshold is defined by the lowest number of reads of non-trashed samples = 1600.

```{r DAT rare, echo=F, message=F, warning=F}

library("vegan")
# rarefication with library(vegan)
bDAT_rare <- (t(rrarefy(t(bDAT_keep), rare_bac)))

#remove ASVs with 0 seqs
bDAT_rare <-as.matrix( bDAT_rare[rowSums(bDAT_rare) > 0,]  )# removal of rows with 0 values

#Repeat for replicate input
inputDAT_rare <- t(rrarefy(t(inputDAT), rare_bac))
#remove ASVs with 0 seqs
inputDAT_rare <- inputDAT_rare[rowSums(inputDAT_rare) > 0,]  # removal of rows with 0 values


#remove famgen color values form color vector for famgen combinations that did not make it beyond the rarefaction step -> no need, we remain wit the same famgen combinations

```


### Figure 3 \| Rarefaction plot 2.0

We plot a rarefaction plot with the remaining samples to check if the sequence depth is enough to capture the microbial diversity.

```{r rarefaction 2.0, eval=T, echo=F, message=F, warning=F, fig.height=6, fig.width=6}

#plot rarefaction
#rarefy

#rarefy
pdf("/dev/null") #prevents drawing of standard rarefaction plot (we will create our own with colors)
brarefaction <- rarecurve(t(bDAT_rare), step=50, 7000, label = FALSE) 
names(brarefaction) <- DESIGN$AccessArray
invisible(dev.off())

# long-transformation.
protox <- mapply(FUN = function(x, y) {
  mydf <- as.data.frame(x)
  colnames(mydf) <- "value"
  mydf$species <- y
  mydf$subsample <- attr(x, "Subsample")
  mydf
}, x = brarefaction, y = as.list(names(brarefaction)), SIMPLIFY = FALSE)

brarefaction_long <- do.call(rbind, protox)
rownames(brarefaction_long) <- NULL  # pretty

#add sample type (root vs sub, sterile vs inocualted)
brarefaction_long$type <- NA
for (i in 1:nrow(brarefaction_long)) {
  brarefaction_long$type[i] <- as.character(DESIGN$type[DESIGN$AccessArray == brarefaction_long$species[i]])
}

#add SynCom9 (yes/no)
brarefaction_long$SynCom9 <- NA
for (i in 1:nrow(brarefaction_long)) {
  brarefaction_long$SynCom9[i] <- as.character(DESIGN$SynCom9[DESIGN$AccessArray == brarefaction_long$species[i]])
}

#add treatment (high/medium/low P)
brarefaction_long$treatment <- NA
for (i in 1:nrow(brarefaction_long)) {
  brarefaction_long$treatment[i] <- as.character(DESIGN$treatment[DESIGN$AccessArray == brarefaction_long$species[i]])
}

#plot
fig3.1 <- ggplot(brarefaction_long, aes(x = subsample, y = value, group = species)) +
                  theme_bw() +
                  geom_line(aes(color=treatment))+
                  scale_color_manual(values = level_cols_treatment,
                                    labels=c(expression("low P"), expression("medium P"),expression("high P")))+
                  xlab("Sequencing Depth")+
                  scale_x_continuous(limits=c(0,1610))+

                  ylab("Number of ASVs")+
                  ggtitle("Rarefaction after cleanup") +
                  theme(legend.position = "bottom",
                        legend.key.height=unit(0.5, 'lines'))

fig3.1
ggsave("fig3.1.png", height=4, width=4) 

fig3.2 <- ggplot(brarefaction_long, aes(x = subsample, y = value, group = species)) +
                  theme_bw() +
                  geom_line(aes(color=type))+
                  scale_color_manual(values = level_cols_type,
                                    labels=c(expression("roots"), expression("sorbix")))+
                  xlab("Sequencing Depth")+
                  scale_x_continuous(limits=c(0,1610))+

                  ylab("Number of ASVs")+
                  ggtitle("Rarefaction after cleanup") +
                  theme(legend.position = "bottom",
                        legend.key.height=unit(0.5, 'lines'))

fig3.2
ggsave("fig3.2.png", height=4, width=4) 
# dev.off()



```

\vspace{5mm}

**Conclusion:** After rarefaction all samples still have sufficient reads?

\pagebreak

## Sample Control

### Sample Size

We end up with the following number of samples per treatment for the analysis.  

```{r sample size rare, warning=F, echo=F}

#final number of samples
table_bac <- table(DESIGN$type, DESIGN$SynCom9, DESIGN$treatment)

#out
pander(table_bac, caption = "Sample profile after cleanup")

```


\pagebreak

```{r export RDA files, echo=F, warning=F, message=F}

# Export

# create directory
#dir.create("../../Results/interim")
#dir.create("../../Results/Sinterim/01_Import_Normalization")

## set output directory
setwd("~/Desktop/MASTER Pflanzenwissenschaften/Experiments/Microbiota analysis/Results/interim/01_Import_Normalization")

#save objects needed in the following scripts
saveRDS(DESIGN, "DESIGN.RDS")
saveRDS(bDAT_keep, "bDAT.RDS")
saveRDS(TAX, "TAX.RDS")
saveRDS(bDAT_rare, "bDAT_rare.RDS")
saveRDS(inputDAT_rare, "inputDAT_rare.RDS")
saveRDS(rare_bac, "rare_bac.RDS")

saveRDS(level_cols_type, "level_cols_type.RDS")
saveRDS(level_cols_treatment, "level_cols_treatment.RDS")
saveRDS(level_cols_SynCom9, "level_cols_SynCom9.RDS")

saveRDS(TAX_level_cols_famgen, "TAX_level_cols_famgen.RDS")
saveRDS(TAX_level_cols_phyla, "TAX_level_cols_phyla.RDS")
saveRDS(TAX_level_cols_SynCom, "TAX_level_cols_SynCom.RDS")

```



